hail/src/main/scala/is/hail/cxx/EmitTriplet.scala

```diff
@@ -75,12 +77,21 @@ class EmitRegion private (val fb: FunctionBuilder, val region: Variable, val poo
   }
 
   def arrayBuilder(fb: FunctionBuilder, pType: PContainer): StagedContainerBuilder = {
-    use(); new StagedContainerBuilder(fb, region.name, pType)
+    use(); new StagedContainerBuilder(fb, region, pType)
   }
 
-  def newRegion(): EmitRegion = new EmitRegion(fb, fb.variable("region", "RegionPtr", s"$pool->get_region()"), pool)
+  def newRegion(): EmitRegion = new EmitRegion(fb, baseRegion, fb.variable("region", "RegionPtr", s"$baseRegion->get_region()"))
 }
 
+object EmitContext {
```

Comment:
I think it would be helpful to use the same name for the scala/c++ representations of the same runtime objects.